lms/resources/_js_config/__init__.py

```diff
@@ -524,7 +524,9 @@ def _sync_api(self, assignment):
                     "product": self._request.product.family,
                 },
                 "context_id": self._request.lti_params["context_id"],
-                "group_set_id": self._context.group_set_id,
+                "group_set_id": self._request.product.plugin.grouping_service.group_set_id(
+                    self._request, assignment
+                ),
```

Comment:
This would be a call to the service instead of the plugin with the suggestion above. Which is the pattern we have elsewhere.

While I'm not opposed to going straight to the plugin, going via the service seems better here. I'll be honest and say this is setting off some non-specific heebie jeebie feeling too. Currently the outside code is not bound to the plugin outside of the service. This feels like it gives us more room for manoeuvre.